ember-element-helper/README.md

```diff
@@ -85,12 +80,93 @@ that accepts "contextual components" as arguments:
 <@tag class="my-tag">hello world!</@tag>
 ```
 
-Contributing
-------------------------------------------------------------------------------
+### Single File Components
+
+Using the `(element)` helper with [first class component
+templates](http://emberjs.github.io/rfcs/0779-first-class-component-templates.html):
+
+```gjs
+import { element } from 'ember-element-helper';
+
+<template>
+  {{#let (element @tagName) as |Tag|}}
+    <Tag class="my-tag">hello world!</Tag>
+  {{/let}}
+</template>
+```
+
+### Glint Usage in Classic Mode
+
+In order to use a typed `(element)` helper in classic mode, you need to import
+the addon's glint template registry and extend your app's registry declaration
+as described in the [Using
+Addons](https://typed-ember.gitbook.io/glint/using-glint/ember/using-addons#using-glint-enabled-addons)
+documentation:
+
+```ts
+import '@glint/environment-ember-loose';
+import type EmberElementHelperRegistry from 'ember-element-helper/template-registry';
+
+declare module '@glint/environment-ember-loose/registry' {
+  export default interface Registry extends EmberElementHelperRegistry, /* other addon registries */ {
+    // local entries
+  }
+}
+```
+
+> **Note:** Glint itself is still under active development, and as such breaking changes might occur.
+> Therefore, Glint support by this addon is also considered experimental, and not covered by our SemVer contract!
+
+### Typing your Components
+
+When your component accepts an element with the `(element)` helper, you want to
+give this argument a proper type. Here is how:
+
+```ts
+import type { ElementSignature } from 'ember-element-helper';
+
+interface YourComponentSignature<T extends string> {
+  Element: HTMLSectionElement;
+  Args: {
+    element?: ElementSignature['Return'];
+  };
+}
+```
+
+When the `@element` argument influences the `Element` of your component:
+
+```ts
+import type { ElementSignature, ElementFromTagName } from 'ember-element-helper';
+
+interface YourComponentSignature<T extends string> {
+  Element: ElementFromTagName<T>;
+  Args: {
+    element?: ElementSignature<T>['Return'];
+  };
+}
+```
+
+When your component already uses an element for a given condition. When
+the condition isn't met, a fallback element is used. The fallback can even be
+provided from the outside. Here is the type:
+
+```ts
+import type { ElementSignature, ElementFromTagName } from 'ember-element-helper';
+
+interface YourComponentSignature<
+  T extends string = 'section'
+> {
+  Element: HTMLButtonElement | HTMLAnchorElement | ElementFromTagName<T>;
+  Args: {
+    element?: ElementSignature<T>['Return'];
+  };
+}
+```
```

Comment:
This is giving me problems. See below: https://github.com/tildeio/ember-element-helper/pull/107/files#r1286107261